pytorch_lightning/plugins/precision/double.py

```diff
@@ -90,3 +91,18 @@ def connect(
     def post_dispatch(self) -> None:
         while len(self.patches) > 0:
             self.patches.pop().teardown()
+
+    @contextlib.contextmanager
+    def tensor_type_context(self) -> Generator:
+        """
+        A context manager to change the default tensor type.
+        See: :meth:`torch.set_default_tensor_type`
+        """
+        torch.set_default_tensor_type(torch.DoubleTensor)
+        yield
+        torch.set_default_tensor_type(torch.FloatTensor)
+
+    train_step_context = tensor_type_context
+    val_step_context = tensor_type_context
+    test_step_context = tensor_type_context
+    predict_step_context = tensor_type_context
```

Comment:
@tchaton this is breaking the plugin typechecks with mypy:
```
pytorch_lightning/plugins/precision/double.py:105: error: Incompatible types in assignment (expression has type "Callable[[], _GeneratorContextManager[Any]]", base class "Plugin" defined the type as "Callable[[], Generator[Any, Any, Any]]")  [assignment]
pytorch_lightning/plugins/precision/double.py:106: error: Incompatible types in assignment (expression has type "Callable[[], _GeneratorContextManager[Any]]", base class "Plugin" defined the type as "Callable[[], Generator[Any, Any, Any]]")  [assignment]
pytorch_lightning/plugins/precision/double.py:107: error: Incompatible types in assignment (expression has type "Callable[[], _GeneratorContextManager[Any]]", base class "Plugin" defined the type as "Callable[[], Generator[Any, Any, Any]]")  [assignment]
pytorch_lightning/plugins/precision/double.py:108: error: Incompatible types in assignment (expression has type "Callable[[], _GeneratorContextManager[Any]]", base class "Plugin" defined the type as "Callable[[], Generator[Any, Any, Any]]")  [assignment]
```

rather than overriding these step contexts, could we share the same implementation?